### ISSUE TITLE
QSocketNotifier error while compiling with pytorch.

### ISSUE THREAD

**Issue by Apisteftos** [pytorch/pytorch#95138]
### 🐛 Describe the bug

I am running this example [https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html](https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html) on Ubuntu in VScode and I am getting error, but while I am running the same example on Windows 10, there is no problem. 

### Error logs

Error:
============
<pre>
QSocketNotifier: Can only be used with threads started with QThread
Exception ignored in: <function TimerQT.__del__ at 0x7f1cd030e200>
Traceback (most recent call last):
  File "/home/me/Desktop/Programming/Python/.venv/lib/python3.10/site-packages/matplotlib/backends/backend_qt.py", line 174, in __del__
TypeError: 'NoneType' object is not callable
</pre>

line 174

` nn.LeakyReLU(0.2, inplace=True) `

I thought it would be a problem from another program `QTcreator5` and I uninstalled it and reboot, but the problem still exists.

### Versions

<pre>
Collecting environment information...
PyTorch version: 1.13.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.2 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-5.19.0-32-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   43 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          12
On-line CPU(s) list:             0-11
Vendor ID:                       AuthenticAMD
Model name:                      AMD Ryzen 5 1600 Six-Core Processor
CPU family:                      23
Model:                           1
Thread(s) per core:              2
Core(s) per socket:              6
Socket(s):                       1
Stepping:                        1
Frequency boost:                 enabled
CPU max MHz:                     3200,0000
CPU min MHz:                     1550,0000
BogoMIPS:                        6399.20
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb hw_pstate ssbd ibpb vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 xsaves clzero irperf xsaveerptr arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif overflow_recov succor smca sev
Virtualization:                  AMD-V
L1d cache:                       192 KiB (6 instances)
L1i cache:                       384 KiB (6 instances)
L2 cache:                        3 MiB (6 instances)
L3 cache:                        16 MiB (2 instances)
NUMA node(s):                    1
NUMA node0 CPU(s):               0-11
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Mitigation; untrained return thunk; SMT vulnerable
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.22.4
[pip3] torch==1.13.1
[pip3] torchaudio==0.13.1+rocm5.2
[pip3] torchvision==0.14.1
[conda] Could not collect
</pre>

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by albanD** [comment]
The error does point at qt code so that looks unrelated to pytorch? Are you sure you're calling the matplotlib API properly here?

**Comment by Apisteftos** [comment]
I fixed that, it was the packages from `sudo apt install PyQT5` and from `.venv` . I removed completely the packages with `sudo apt remove PyQT5` and also I removed from .venv with `pip uninstall PyQT5` and then I reinstalled only in pip. PyQT5 is a package in matplotlib on Linux for plotting.  

I get this now: 
<pre>
QSocketNotifier: Can only be used with threads started with QThread
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()

</pre>

I think is just a warning and nothing important.
